### PR TITLE
feat: kubernetes keycloak deployment strategy

### PR DESCRIPTION
**Pull request by kesslerdev**
### What does this PR do?
Allow to set the deployment strategy, for the keycloak deployment chart
the default strategy RollingUpdate fail if multiattach is not supported by volumes

### CODE REVIEW

File: deploy/kubernetes/helm/che/custom-charts/che-keycloak/values.yaml [hunk 1]
```diff
@@ -15,3 +15,4 @@ image: quay.io/eclipse/che-keycloak:nightly
 requireAdminPasswordChange: true
 keycloakAdminUserName: admin
 keycloakAdminUserPassword: admin
+deploymentStrategy: RollingUpdate
```
Comment (sleshchenko) [inline deploy/kubernetes/helm/che/custom-charts/che-keycloak/values.yaml]:
```suggestion
deploymentStrategy: RollingUpdate

```

Comment (sleshchenko) [inline deploy/kubernetes/helm/che/custom-charts/che-keycloak/values.yaml]: > Allow to set the deployment strategy, the default RollingUpdate fail if multiattach is not supported by volumes

I think in most cases it won't be supported. What about setting default to recreate which mean - introduce an ability to override default but not change the behavior we already have?

Comment (kesslerdev) [reply deploy/kubernetes/helm/che/custom-charts/che-keycloak/values.yaml]: yes

Comment (kesslerdev) [reply deploy/kubernetes/helm/che/custom-charts/che-keycloak/values.yaml]: but other charts postgres, che, already have Recreate as deploy strategy

**Review by mmorhun — APPROVED** [review]

**Comment by tolusha** [comment]
[test]

**Comment by dmytro-ndp** [comment]
[ci-build]

**Comment by dmytro-ndp** [comment]
[crw-ci-test]

**Comment by dmytro-ndp** [comment]
@tolusha: commands to trigger CI jobs:
```
[ci-build] - ci.centos.org PR check (build)
[ci-test] - ci.centos.org PR check (e2e tests)
[crw-ci-test] - crw-ci-e2e-happy-path-tests 
```

**Comment by tolusha** [comment]
[crw-ci-test]

**Comment by tolusha** [comment]
[crw-ci-test]

**Comment by mmorhun** [comment]
cc @dmytro-ndp could you please take a look why the tests failed? Looks like not PR related. Thanks.

**Comment by mmorhun** [comment]
[crw-ci-test]

**Comment by Ohrimenko1988** [comment]
[crw-ci-test]

**Comment by Ohrimenko1988** [comment]
[crw-ci-test]

**Comment by flacatus** [comment]
[crw-ci-test]

**Comment by flacatus** [comment]
[crw-ci-test]

**Comment by dmytro-ndp** [comment]
@kesslerdev: PR needs to be updated from master branch, to fix Happy path tests build error:
```
[Build Che including upstream projects] [ERROR] The build could not read 1 project -> [Help 1]
[Build Che including upstream projects] [ERROR]   
[Build Che including upstream projects] [ERROR]   The project org.eclipse.che.docs:che-docs:7.13.0-SNAPSHOT (/mnt/hudson_workspace/workspace/Multiuser-Che-PR-check-E2E-Happy-path-tests-against-k8s/eclipse/che-docs/pom.xml) has 1 error
[Build Che including upstream projects] [ERROR]     Non-resolvable parent POM for org.eclipse.che.depmgt:maven-depmgt-pom:7.13.0-SNAPSHOT: Could not find artifact org.eclipse.che.parent:maven-parent-pom:pom:7.13.0-SNAPSHOT in ossrh (https://oss.sonatype.org/content/repositories/snapshots/) @ org.eclipse.che.depmgt:maven-depmgt-pom:7.13.0-SNAPSHOT, /home/hudson/.m2/repository/org/eclipse/che/depmgt/maven-depmgt-pom/7.13.0-SNAPSHOT/maven-depmgt-pom-7.13.0-SNAPSHOT.pom, line 17, column 13 -> [Help 2]
```

**Comment by mmorhun** [comment]
@kesslerdev please rebase your changes. Then we can merge this PR.

**Comment by kesslerdev** [comment]
> @kesslerdev please rebase your changes. Then we can merge this PR.

Done !

**Comment by tolusha** [comment]
https://github.com/eclipse/che-operator/pull/449/

**Comment by tolusha** [comment]
[crw-ci-test]

**Comment by flacatus** [comment]
[crw-ci-test]

**Comment by tolusha** [comment]
The only failed tests is related java ls which is a known issue.
The PR is ready to merge.

**Comment by benoitf** [comment]
[ci-build]

**Comment by mmorhun** [comment]
@kesslerdev thank you for contribution.

**Pull request merged**